src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs

```diff
@@ -742,6 +742,77 @@ public class C
 
                 await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, expected: expected, signaturesOnly: signaturesOnly, languageVersion: "Preview", metadataLanguageVersion: "Preview");
             }
+
+            [Theory, CombinatorialData, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+            public async Task TestRequiredProperty(bool signaturesOnly)
+            {
+                var metadataSource = """
+                    public class C { public required int Property { get; set; } }
+                    namespace System.Runtime.CompilerServices
+                    {
+                        public sealed class RequiredMemberAttribute : Attribute { }
+                        public sealed class CompilerFeatureRequiredAttribute : Attribute
+                        {
+                            public CompilerFeatureRequiredAttribute(string featureName)
+                            {
+                                FeatureName = featureName;
+                            }
+                            public string FeatureName { get; }
+                            public bool IsOptional { get; set; }
+                        }
+                    }
+
+                    """;
+                var symbolName = "C";
+
+                // ICSharpDecompiler does not yet support decoding required members nicely
+                var expected = signaturesOnly switch
+                {
+                    true => $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+using System.Runtime.CompilerServices;
+
+[RequiredMember]
+public class [|C|]
+{{
+    public C();
+
+    [RequiredMember]
```

Comment:
Do we have a tracking bug for filtering these out somewhere?

Reply:
https://github.com/dotnet/roslyn/issues/61511